src/profiling/with_perf.md

```diff
@@ -0,0 +1,294 @@
+# Profiling with perf
+
+sThis is a guide for how to profile rustc with perf.
+
+## Initial steps
+
+- Get a clean checkout of rust-lang/master, or whatever it is you want to profile.
+- Set the following settings in your `config.toml`:
+  - `debuginfo-lines = true`
+  - `use-jemalloc = false` -- lets you do memory use profiling with valgrind 
+  - leave everything else the defaults
+- Run `./x.py build` to get a full build
+- Make a rustup toolchain (let's call it `rust-prof`) pointing to that result
+  - `rustup toolchain link` XXX
```

Comment:
Sorry, what is `XXX` here?

Reply:
probably me just not remembering that crazy syntax

Reply:
:stuck_out_tongue: Perhaps link to `how-to-build-and-run.html#creating-a-rustup-toolchain`